src/domain/core/src/services/push_service.rs

```diff
@@ -29,6 +30,11 @@ pub trait PushService: Send + Sync {
         options: PushMultiOptions,
         sync_listener: Option<Arc<dyn SyncMultiListener>>,
     ) -> Vec<PushResponse>;
+    /// Returns sync status of all push remotes connected with a given dataset
+    async fn status_multi(
```

Comment:
We use `_multi` prefix for methods that operate on multiple datasets at a time, but here you are working with just one. I would rename to `check_remote_status` to highlight that we're actually reaching out to remote repos here.

Reply:
done